packages/ui/src/components/va-form/VaForm.stories.ts

```diff
@@ -1,12 +1,443 @@
+        :stateful="true"
+        label="save value"
+      />
+    </va-form>
+    [default]
+    <va-form class="flex flex-col max-w-sm">
+      <va-input 
+        :stateful="false"
+        label="does not save value"
+      />
+    </va-form>
+  `,
+})
+
+Stateful.play = async ({ canvasElement, step }) => {
+  const canvas = within(canvasElement)
+  const firstInput = canvas.getByRole('textbox', { name: 'save value' }) as HTMLElement
+  const secondInput = canvas.getByRole('textbox', { name: 'does not save value' }) as HTMLElement
+
+  await step('Should save input data', async () => {
+    await userEvent.type(firstInput, 'data')
+    expect(firstInput).toHaveValue('data')
+  })
+
+  await step('Should not save input data', async () => {
+    await userEvent.type(secondInput, 'data')
+    expect(secondInput).toHaveValue('')
+    secondInput.blur()
+  })
+}
+
+export const ValidationRules = () => ({
+  components: { VaForm, VaInput, VaButton },
+  data: () => ({ input: 'hell' }),
+  template: `
+    [immediate]
+    <va-form
+      class="flex max-w-sm"
+      immediate
+    >
+      <va-input
+        v-model="input"
+        label="immediate validation"
+        :rules="[value => value === 'hello' || 'should be hello']"
+      />
+    </va-form>
+    [deffault]
+    <va-form
+      class="flex max-w-sm"
+    >
+      <va-input
+        v-model="input"
+        label="validation on change"
+        :rules="[value => value === 'hello' || 'should be hello']"
+      />
+    </va-form>
+  `,
+})
+
+ValidationRules.play = async ({ canvasElement, step }) => {
+  const canvas = within(canvasElement)
+  const firstInput = canvas.getByRole('textbox', { name: 'immediate validation' }) as HTMLElement
+  const secondInput = canvas.getByRole('textbox', { name: 'validation on change' }) as HTMLElement
+
+  await step('Displays error message', async () => {
+    expect(firstInput.getAttribute('aria-invalid')).toEqual('true')
+  })
+
+  await step('Does not display error message', async () => {
+    expect(secondInput.getAttribute('aria-invalid')).toEqual('false')
+  })
+}
+
+export const HideErrorMessages = () => ({
+  components: { VaForm, VaInput, VaButton },
+  data: () => ({ input: 'error' }),
+  template: `
+    [hidden]
+    <va-form
+      class="flex max-w-sm"
+      ref="rulesFormRef"
+      immediate
+      :hideErrorMessages="true"
+    >
+      <va-input
+        v-model="input"
+        label="with no error message"
+        :rules="[value => 'error']"
+      />
+    </va-form>
+    [default]
+    <va-form
+      class="flex max-w-sm"
+      ref="rulesFormRef"
+      immediate
+      :hideErrorMessages="false"
+    >
+      <va-input
+        v-model="input"
+        label="with error message"
+        :rules="[value => 'error']"
+      />
+    </va-form>
+  `,
+})
+
+HideErrorMessages.play = async ({ step }) => {
+  const [firstDefinedInput, secondDefineInput] = document.querySelectorAll('div.va-input-wrapper > div.va-message-list')
```

Comment:
I would suggest you either check for text directly or use aria-role. Classes are mostly for style and might change in unpredictable ways. Role stays stable. Text also does (if you define it).